src/transport/message_handler.rs

```diff
@@ -0,0 +1,74 @@
+use crate::transport::{jsonrpc, msgs};
+use bitcoin::secp256k1::PublicKey;
+use lightning::ln::{peer_handler::CustomMessageHandler, wire::CustomMessageReader};
+use std::{io, sync::Mutex};
+
+struct LSPSMessageHandler {
+	pending_messages: Mutex<Vec<(PublicKey, msgs::LSPSMessage)>>,
+}
+
+impl LSPSMessageHandler {
+	fn handle_lsps_request(
+		&self, _request: jsonrpc::Request, _sender_node_id: &PublicKey,
+	) -> Result<(), lightning::ln::msgs::LightningError> {
+		Ok(())
+	}
+
+	fn handle_lsps_response(
+		&self, _response: jsonrpc::Response, _sender_node_id: &PublicKey,
+	) -> Result<(), lightning::ln::msgs::LightningError> {
+		Ok(())
+	}
+
+	fn handle_lsps_notification(
+		&self, _notification: jsonrpc::Notification, _sender_node_id: &PublicKey,
+	) -> Result<(), lightning::ln::msgs::LightningError> {
+		Ok(())
+	}
+
+	fn handle_lsps_message(
+		&self, msg: msgs::LSPSMessage, sender_node_id: &PublicKey,
+	) -> Result<(), lightning::ln::msgs::LightningError> {
+		match msg {
+			msgs::LSPSMessage::Request(request) => {
+				self.handle_lsps_request(request, sender_node_id)
+			}
+			msgs::LSPSMessage::Response(response) => {
+				self.handle_lsps_response(response, sender_node_id)
+			}
+			msgs::LSPSMessage::Notification(notification) => {
+				self.handle_lsps_notification(notification, sender_node_id)
+			}
+		}
+	}
+}
+
+impl CustomMessageReader for LSPSMessageHandler {
+	type CustomMessage = msgs::LSPSMessage;
+
+	fn read<R: io::Read>(
+		&self, message_type: u16, buffer: &mut R,
+	) -> Result<Option<Self::CustomMessage>, lightning::ln::msgs::DecodeError> {
+		match message_type {
+			37913 => {
+				let msg: msgs::LSPSMessage = serde_json::from_reader(buffer)
+					.map_err(|_| lightning::ln::msgs::DecodeError::InvalidValue)?;
```

Comment:
There's a bit in the spec about returning jsonrpc::ErrorResponse if we receive an invalid request.  I'm not quite sure how we can do that given this interface.

In theory I'd like to queue the msg here.

Makes me think that we will need to read it as a raw String and handle it up-a-level in the message handler, right?
We can attempt deserialization and then have access to the pending message queue there.